### ISSUE TITLE
NW.js does not start when Avast antivirus is running

### ISSUE THREAD

**Issue by ValeryEliseev** [nwjs/nw.js#4023]
nwjs-v0.13.0-alpha7-win-x64 does not start on Windows 8.1 when Avast Free Antivirus 10.4.2233 is running. If Avast stop nwjs run normally.

**Comment by Mithgol** [comment]
Does Avast report anything in its log? (Does it have a false positive, perhaps?)

**Comment by ValeryEliseev** [comment]
Avast does not report nwjs as a virus or suspicious program. Nwjs just "not run", silent. No error, no warning, no nw.exe process in windows tasks. Nothing! 

This problem not appears when switch off Avast "DeepScreen" option (checked by default).

My Avast logs example with matches "nw.exe" pattern: 

C:\Users\All Users\AVAST Software\Avast\log\aswAr.log 
---
Process E:\nwjs-v0.13.0-alpha7-win-x64\nw.exe [5208]
---

C:\Users\All Users\AVAST Software\Avast\log\autosandbox.log
---
02.12.2015 22:05:47	Autosandbox candidate: E:\nwjs-v0.13.0-alpha7-win-x64\nw.exe
	[Source: local://*C:\Windows\explorer.exe		]
	[Opened by: E:\nwjs-v0.13.0-alpha7-win-x64\nw.exe]
	[Reason: 0x00020000]
	 --> Result: Sandboxing (NG component is not installed)
	 --> Instrumentation: Instrumentation inside sandbox requested
---

Similar Avast problem https://forum.avast.com/index.php?topic=178961.0

P.S. Perhaps it's Avast problem, but why with nwjs?

**Comment by Mithgol** [comment]
Looking at the logs you've provided I am tempted to believe that Avast tries to sandbox nw.exe and that sandboxing went wrong.

Why it does that with a Chromium-based browser? — perhaps because authors of some malware try to install their own Chromium-based browsers on the attacked systems (as reported by [Malwarebytes](https://blog.malwarebytes.org/online-security/2015/10/efast-browser-hijacks-file-associations/); you might have already read about it on [GeekTimes](http://geektimes.ru/post/264164/)).

You should report this problem to Avast and expect that they (as in https://forum.avast.com/index.php?topic=178961.0 before) might ask you to check the new alpha of NW.js on VirusTotal. (You could unknownly have your nw.exe infected, after all.) When you report, post an URL (of a forum thread) here because it would help Avast's users that encounter the same issue but later than you.

Recently I was relatively busy IRL (busier than before August, for example) and thus I haven't yet made myself familiar with the latest version of Electron. That's why I won't plan to post anything on Habrahabr about Electron in the nearest future. (Things might change after we meet the new year and celebrate the usual ten days of national holidays, but I won't promise that because I might spend these days on something else entirely.)

**Comment by makkesk8** [comment]
Experiencing the same issue as well... disabling avast works. However I only have this issue when compiling my app into a .exe so "debug" mode works just fine.

Same scenario at work were we are using Norton. Seems like a huge issue atm.

**Comment by Mithgol** [comment]
These “compiled” NW.js applications are ZIP packages attached to the tail of the engine's EXE file.

Could it be that antiviruses simply treat anything behind the expected end of EXE files as suspicious?

@ValeryEliseev Have you also experienced you problem only for packed application(s)? Or unpacked as well?

**Comment by makkesk8** [comment]
@Mithgol yes, this could be the case.. It crossed my mind. However, there's nothing out there available that does this for free and is compatible with nw.js.

Somebody please prove me wrong, I'd hate to ask our network administrator to whitelist my apps.

**Comment by MarcFowler** [comment]
Did anyone have any luck with this problem? I'm encountering the same thing. No threats from virus scan but Avast prevents it from loading.

**Comment by friedolinfoerder** [comment]
What's the state/progress of this issue?
I've got the same behaviour: The nw-application does not start with activated avast virus scanner

**Comment by Christywl** [comment]
I can reproduce this issue on Windows with nwjs-sdk-v0.25.0.
